File: java-service/src/main/java/com/poc/tutorials/model/TutorialEntity.java
package com.poc.tutorials.model;

import lombok.*;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.util.UUID;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor

@Table(schema = "poc", name = "tutorial")
public class TutorialEntity {

  @Id
  @GeneratedValue(generator = "UUID")
  @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
  @Column(
    name = "id",
    nullable = false,
    insertable = false,
    updatable = false,
    columnDefinition = "UUID"
  )
  private UUID id;

  @Column(name = "title", nullable = false)
  private String title;

  @Column(name = "description")
  private String description;

  @Column(name = "published")
  private boolean published;


  public TutorialEntity(String title, String description, boolean published) {
    this.title = title;
    this.description = description;
    this.published = published;
  }

  public UUID getId() {
    return id;
  }

  public String getTitle() {
    return title;
  }

  public void setTitle(String title) {
    this.title = title;
  }

  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public boolean isPublished() {
    return published;
  }

  public void setPublished(boolean isPublished) {
    this.published = isPublished;
  }
}

File: kotlin-service/src/main/kotlin/com/example/poc/model/TutorialEntity.kt
package com.example.poc.model

import jakarta.persistence.*
import org.hibernate.annotations.GenericGenerator
import java.util.UUID


@Entity(name = "tutorial")
@Table(schema = "tutorials", name = "tutorial")
data class TutorialEntity(
        @Id
        @GeneratedValue(generator = "UUID")
        @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
        @Column(
                name = "id",
                nullable = false,
                insertable = false,
                updatable = false,
                columnDefinition = "UUID"
        )
        val id: UUID = UUID.randomUUID(),
        @Column(name = "title", nullable = false)
        val title: String = "",
        @Column(name = "description")
        val description: String = "",
        @Column(name = "published")
        val published: Boolean = false, ) {}
